RUFAS/routines/animal/ration/ration_optimizer.py

```diff
@@ -897,7 +898,7 @@ def make_user_bounds(
     def optimize(
         self, animal_combination: AnimalCombination,
         ration_config: RationConfig,
-        previous_ration: Dict[str, float | str] | None = None
+        previous_ration: Mapping[str, float | str] | None = None
```

Comment:
```suggestion
        previous_ration: Dict[str, float | str] | None = None
```
Instead of changing this to `Mapping`, I suggest keeping it a `Dict` and adding the following at line 934: `value = int(value)`. The benefit is that it keeps the typing of `previous_ration` more consistent with the way it is typed in other places, and doesn't add the error associated with `Mapping`.

Reply:
Good point, thank you! I think adding Mapping removed a Mypy error in the test_animal.py file but I did wince a bit when I added it.